cli/command/container/run.go

```diff
@@ -182,9 +182,19 @@ func runContainer(dockerCli *command.DockerCli, opts *runOptions, copts *contain
 		if err != nil {
 			return err
 		}
-	}
 
-	statusChan := waitExitOrRemoved(ctx, dockerCli, createResponse.ID, copts.autoRemove)
+		statusChan := waitExitOrRemoved(ctx, dockerCli, createResponse.ID, copts.autoRemove)
+		defer func() {
```

Comment:
The old code seems to skip this if the user enters the detach escape sequence, or if only stdin is attached. Is the change intentional?